### PR TITLE
Feature: Added option to restore all items in recycle bin

### LINKED ISSUE #9561: Bug: Restore option is missing from recycle bin
### Description

Right clicking in the recycle bin there only copy, delete and properties.

**Requirements**
- selecting an item in recycle bin adds a "Restore" button in the toolbar
- selecting multiple items adds a "Restore" button to the toolbar
- if no items are selected, there should toolbar button that says "Restore all items"
- right clicking an item/items adds a "Restore" option to the context menu
- right clicking in a blank space adds a "Restore all items" option to the context menu

### Files Version

2.3.1.0

### Windows Version

10.0.19043.1806

### Relevant Assets/Logs

[debug.log](https://github.com/files-community/Files/files/9105826/debug.log)
[debug_fulltrust.log](https://github.com/files-community/Files/files/9105827/debug_fulltrust.log)

### PR DESCRIPTION
**Pull request by QuaintMako**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Closes #9561 

**Details of Changes**
- Added Restore Recycle Bin action in contextual menu of Recycle Bin when no object is selected.
- Added Restore Recycle Bin action in Inner Navigation Toolbar.
- Added localization linked to the new action.

**Validation**
How did you test these changes?
- [X] Built and ran the app

**Screenshots**

9561_Proof

### CODE REVIEW

**Comment by yaichenbaum** [comment]
> Added Restore Recycle Bin action in contextual menu of Recycle Bin when no object is selected.

I'm pretty sure it's only meant to show when selecting items

> Added Restore Recycle Bin action in Inner Navigation Toolbar.

Great idea! Perhaps it can be simplified to say "restore"?

File: src/Files.Uwp/UserControls/InnerNavigationToolbar.xaml [hunk 1]
```diff
@@ -205,6 +205,18 @@
...
+                <AppBarButton
+                    x:Name="RestoreRecycleBinButton"
+                    x:Load="{x:Bind ViewModel.InstanceViewModel.IsPageTypeRecycleBin, Mode=OneWay, FallbackValue=False}"
+                    AccessKey="B"
+                    Command="{x:Bind ViewModel.RestoreRecycleBinCommand, Mode=OneWay}"
+                    IsEnabled="{x:Bind ViewModel.CanRestoreRecycleBin, Mode=OneWay}"
+                    Label="{helpers:ResourceString Name=RestoreRecycleBin}"
+                    ToolTipService.ToolTip="{helpers:ResourceString Name=RestoreRecycleBin}">
+                    <AppBarButton.Icon>
+                        <FontIcon Glyph="&#xE10D;" />
```
Comment (yaichenbaum) [inline src/Files.Uwp/UserControls/InnerNavigationToolbar.xaml:217]:
```suggestion
                        <FontIcon Glyph="&#xE845;" />
```

**Comment by QuaintMako** [comment]
I probably won't be able to attend to the PR for a few more days. If someone wants to take it up, feel free. 
Will try to do it on the week-end.

**Comment by QuaintMako** [comment]
> > Added Restore Recycle Bin action in Inner Navigation Toolbar.
> 
> Great idea! Perhaps it can be simplified to say "restore" when multiple items are selected? I'd add a separate button when no items are selected that says "Restore all items".

Working on it.

**Comment by QuaintMako** [comment]
@yaichenbaum you're asking for two different buttons to the restore action in the navigation toolbar. One to restore files and one to restore the selected items.

Should I implement two similar buttons for the emptying? One to empty all, one to empty the selection.

**Comment by yaichenbaum** [comment]
> Should I implement two similar buttons for the emptying? One to empty all, one to empty the selection.

That's correct, the first option should only be visible when nothing is selected.

**Comment by yaichenbaum** [comment]
@QuaintMako can you resolve the merge conflict?

**Comment by QuaintMako** [comment]
@yaichenbaum the PR is still being worked on, I've still got some to do to finish to add the action to empty the selection from the recycle bin.

Although it seems the PR has some issues I did not see. It's not going as smoothly as it should. Maybe I should close this PR for now and open a new one when I'm done?

**Comment by yaichenbaum** [comment]
Happy for you to leave it open as a draft 👍

**Comment by QuaintMako** [comment]
Learning the existence of that draft thingy, sweet! Converting it into one.

**Comment by QuaintMako** [comment]
Closing the PR to redo it entirely. 
Some of the part used by the made work are using parts that have been deeply changed by new contributions.  Will be easier and cleaner to start from an updated state.

**Pull request closed without merging**